core/src/test/java/com/linecorp/armeria/client/ConnectionPoolCollectingMetricTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static com.linecorp.armeria.common.HttpStatus.OK;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.metric.MoreMeters;
+import com.linecorp.armeria.common.metric.PrometheusMeterRegistries;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+public class ConnectionPoolCollectingMetricTest {
+
+    private ConnectionPoolListener connectionPoolListener;
+    final MeterRegistry registry = PrometheusMeterRegistries.newRegistry();
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.idleTimeoutMillis(0);
+            sb.requestTimeoutMillis(0);
+            sb.service("/", (ctx, req) -> HttpResponse.of(OK));
+        }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
+    };
+
+    @RegisterExtension
+    static ServerExtension server2 = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.idleTimeoutMillis(0);
+            sb.requestTimeoutMillis(0);
+            sb.service("/", (ctx, req) -> HttpResponse.of(OK));
+        }
+
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
```

Comment:
Is it necessary to restart the server each time a test is run?   I think test time can be reduced significantly if this is removed. Ditto for `server` as well

Reply:
Yes, it is a waste. We dont need to restart the server each time a test is run. Removed and it helps the test run 3x faster. Thank you so much